File: Code/BeforeLegends/Assets/Scripts/World Map/Pathfinding/PathNode.js
//#pragma strict
//import System.Collections.Generic;

//public class PathNode{
//	var g : float = 0;
//	var h : float = 0;
//	var f : float = 0;

//	var x : int;
//	var y : int;

//	var parent : PathNode = null;
	
//	function PathNode(x : int, y : int){
//		this.x = x;
//		this.y = y;
//	}
	
//	function PathNode(x : int, y : int, parent : PathNode, goalX : int, goalY : int){
//		this.x = x;
//		this.y = y;
//		this.parent = parent;
//		evalCost(goalX, goalY);
//	}
	
//	function evalCost(goalX : int, goalY : int){
//		var data : WorldMapData = WorldMapData.getInstance();
//		g = parent.g + 1;
//		var goal : Vector3 = data.tiles[goalX, goalY].position;
//		var current : Vector3 = data.tiles[x, y].position;
//		h = Mathf.Abs(goal.x - current.x) + Mathf.Abs(goal.z - current.z);	
//		f = g + h;
//	}
	
//	function tryAlternative(alt : PathNode){
//		var altG : float = alt.g + 1;
//		if(altG < g){
//			parent = alt;
//			g = altG;
//			f = g + h;
//		}
//	}
	
//	function toPath() : Vec2i[]{
//		var path : Stack.<Vec2i> = new Stack.<Vec2i>();
//		var current : PathNode = this;
//		while(current){
//			path.Push(Vec2i(current.x, current.y));
//			current = current.parent;
//		}
//		return path.ToArray();
//	}
//}

File: Code/BeforeLegends/Assets/Scripts/World Map/Pathfinding/PathNode.cs
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

public class PathNode{

    public float g = 0;
    public float h = 0;
	public float f = 0;

    public int x;
    public int y;

    PathNode parent = null;



    public PathNode(int x = 0, int y = 0, PathNode parenti = null, int goalX = 0, int goalY = 0)
    {
		this.x = x;
		this.y = y;
        if (parenti != null)
        {
		    this.parent = parenti;
		    evalCost(goalX, goalY);
        }
	}
	
	void evalCost(int goalX, int goalY){
		WorldMapData data = WorldMapData.instance;
		g = parent.g + 1;
		Vector3 goal = data.tiles[goalX, goalY].position;
		Vector3 current = data.tiles[x, y].position;
		h = Mathf.Abs(goal.x - current.x) + Mathf.Abs(goal.z - current.z);	
		f = g + h;
	}
	
	public void tryAlternative(PathNode alt){
		float altG = alt.g + 1;
		if(altG < g){
			parent = alt;
			g = altG;
			f = g + h;
		}
	}
	
	 public Vec2int[] toPath(){
		Stack<Vec2int> path = new Stack<Vec2int>();
		PathNode current = this;
		while(current != null){
			path.Push(new Vec2int(current.x, current.y));
			current = current.parent;
		}
		return path.ToArray();
	}
}
